tests/suites/test_suite_ssl.function

```diff
@@ -3300,373 +3568,129 @@ exit:
-#if defined(MBEDTLS_KEY_EXCHANGE__SOME__PSK_ENABLED)
-    const char *psk_identity = "foo";
-#else
-    (void) psk_str;
-#endif
-#if defined(MBEDTLS_TIMING_C)
-    mbedtls_timing_delay_context timer_client, timer_server;
-#endif
-#if defined(MBEDTLS_SSL_CONTEXT_SERIALIZATION)
-    enum { MSGLEN = 5 };
-    unsigned char *context_buf = NULL;
-    size_t context_buf_len;
-    unsigned char cli_buf[MSGLEN];
-    unsigned char srv_buf[MSGLEN];
-#else
-    (void) serialize;
-#endif
-
-    mbedtls_test_message_queue server_queue, client_queue;
-    mbedtls_test_message_socket_context server_context, client_context;
-
-    /* Client side */
-    if( dtls != 0 )
-    {
-        TEST_ASSERT( mbedtls_endpoint_init( &client, MBEDTLS_SSL_IS_CLIENT,
-                                            pk_alg, &client_context,
-                                            &client_queue,
-                                            &server_queue ) == 0 );
-#if defined(MBEDTLS_TIMING_C)
-        mbedtls_ssl_set_timer_cb( &client.ssl, &timer_client,
-                                  mbedtls_timing_set_delay,
-                                  mbedtls_timing_get_delay );
-#endif
-    }
-    else
-    {
-        TEST_ASSERT( mbedtls_endpoint_init( &client, MBEDTLS_SSL_IS_CLIENT,
-                                            pk_alg, NULL, NULL, NULL ) == 0 );
-    }
-    mbedtls_ssl_conf_min_version( &client.conf, MBEDTLS_SSL_MAJOR_VERSION_3,
-                                      version );
-    mbedtls_ssl_conf_max_version( &client.conf, MBEDTLS_SSL_MAJOR_VERSION_3,
-                                      version );
+    handshake_test_options options;
+    init_handshake_options( &options );
 
-    if( strlen( cipher ) > 0 )
-    {
-        set_ciphersuite( &client.conf, cipher, forced_ciphersuite );
-    }
-    /* Server side */
-    if( dtls != 0 )
-    {
-        TEST_ASSERT( mbedtls_endpoint_init( &server, MBEDTLS_SSL_IS_SERVER,
-                                            pk_alg, &server_context,
-                                            &server_queue,
-                                            &client_queue) == 0 );
-#if defined(MBEDTLS_TIMING_C)
-        mbedtls_ssl_set_timer_cb( &server.ssl, &timer_server,
-                                  mbedtls_timing_set_delay,
-                                  mbedtls_timing_get_delay );
-#endif
-    }
-    else
+    options.version = version;
+    options.dtls = dtls;
+    /* Note - the case below will have to updated, since the test sends no data
+     * due to a 1n-1 split against BEAST, that was not expected when preparing
+     * the fragment counting code. */
+    if( version == MBEDTLS_SSL_MINOR_VERSION_0 ||
+        version == MBEDTLS_SSL_MINOR_VERSION_1 )
     {
-        TEST_ASSERT( mbedtls_endpoint_init( &server, MBEDTLS_SSL_IS_SERVER,
-                                            pk_alg, NULL, NULL, NULL ) == 0 );
+        options.cli_msg_len = 0;
+        options.srv_msg_len = 0;
```

Comment:
To keep testing the app data maybe we could also use `expected_cli_fragments *= 2;`, but I'm not sure. Anyway, we have this issue added for the future works